tests/shared/core/test_domain.py

```diff
@@ -1320,3 +1322,118 @@ def test_is_valid_domain_doesnt_raise_with_invalid_yaml(tmpdir: Path):
         potential_domain_path,
     )
     assert not Domain.is_domain_file(potential_domain_path)
+
+
+@pytest.mark.parametrize(
+    "domain_as_dict, expected_not_intent_value",
```

Comment:
since only `"not_intent"` and `GLOBAL_NOT_INTENT` change, you could just paramatize those values, making the whole thing much more succinct.